snakemake/__init__.py

```diff
@@ -1567,6 +1570,11 @@ def get_argument_parser(profile=None):
         action="store_true",
         help="Do not remove incomplete output files by failed jobs.",
     )
+    group_utils.add_argument(
+        "--drop-metadata",
+        action="store_true",
+        help="Drop metadata after job completion.",
```

Comment:
Please also mention the missing provenance information this leads to here (for --report, --list-params-changes, etc.).